### ISSUE TITLE
New data in minecraft-data

### ISSUE THREAD

**Issue by rom1504** [PrismarineJS/minecraft-data#6]
There's some more data that I think can go from mineflayer to minecraft-data : 
* [ ] animationEvents / entityStatusEvents / spawnedObjectTypes / mobTypes https://github.com/andrewrk/mineflayer/blob/master/lib/plugins/entities.js#L10 https://github.com/andrewrk/mineflayer/issues/207
* [ ] NAMED_ENTITY_HEIGHT / CROUCH_HEIGHT https://github.com/andrewrk/mineflayer/blob/master/lib/plugins/entities.js#L5
* [ ] dimension_names https://github.com/andrewrk/mineflayer/blob/master/lib/plugins/game.js#L3
* [ ] difficultyNames https://github.com/andrewrk/mineflayer/blob/master/lib/plugins/game.js#L9
* [ ] gameModes https://github.com/andrewrk/mineflayer/blob/master/lib/plugins/game.js#L46
* [ ] physics constants https://github.com/andrewrk/mineflayer/blob/master/lib/plugins/physics.js#L16
* [ ] viewDistanceToBits https://github.com/andrewrk/mineflayer/blob/master/lib/plugins/settings.js#L11
* [ ] armorSlots https://github.com/andrewrk/mineflayer/blob/master/lib/plugins/simple_inventory.js#L9
* [ ] QUICK_BAR_COUNT and QUICK_BAR_START https://github.com/andrewrk/mineflayer/blob/master/lib/plugins/simple_inventory.js#L6
* [ ] INVENTORY_SLOT_COUNT https://github.com/andrewrk/mineflayer/blob/master/lib/windows.js#L7
* [ ] windows https://github.com/andrewrk/mineflayer/blob/master/lib/windows.js#L22 : some change in mineflayer code needed for this, the data is at least : window names, inventorySlotStart

See what kind of .json this would go into.

**Comment by rom1504** [comment]
maybe these kind of thing too https://github.com/AlexKvazos/MinecraftChat/blob/master/app/server/parsers/vanilla.js

**Comment by rom1504** [comment]
spawnedObjectTypes / mobTypes are integrated (entities.json)

**Comment by Gjum** [comment]
Wurstmineberg has a decent collection: https://github.com/wurstmineberg/assets.wurstmineberg.de/tree/master/json

**Comment by nickelpro** [comment]
Those Physics constants are only true for Mineflayer's implementation of Physics which is divergent from the 20tps notchian standard. Physics constants should probably come from the wiki

http://minecraft.gamepedia.com/Transportation
http://minecraft.gamepedia.com/Gravity#Motion_of_entities

**Comment by rom1504** [comment]
no, mineflayer plugins run at 20tps https://github.com/andrewrk/mineflayer/blob/master/lib/plugins/physics.js#L12
(the position update run at 50tps but that's a separate thing)

Now minecraft-data should probably use the /tick unity yes.

**Comment by rom1504** [comment]
@Gjum yeah I saw that. Not sure what we need from what they have, not sure if taking what they have is a correct extraction process (what happens when they stop updating ?)

**Comment by nickelpro** [comment]
@rom1504 You have your calculation backwards
Physics runs every 20ms, 1sec/20ms is 50tps
Position Update runs every 50ms, 1sec/50ms is 20tps

Which makes perfect sense, if Mineflayer was updating at 50tps it would get hungry and die very quickly. Notchian client runs Physics and Player Update at the same time, 20tps. I never understood why Mineflayer ran Physics so fast.

**Comment by rom1504** [comment]
Ah indeed you're right.
I don't know why we are running physics so fast, I think at some point I tried to change it to the same value as position and it didn't change much, I might try that again.
But anyway, mineflayer takes into account the deltaSecond before the last iteration (https://github.com/andrewrk/mineflayer/blob/master/lib/plugins/physics.js#L127), so our values are independent of the ticking speed.

**Comment by Gjum** [comment]
Please add wiki.vg/Entities#Mobs, as this is not the same as in #11.

**Comment by rom1504** [comment]
Seems both page have pbs (http://minecraft.gamepedia.com/Data_values/Entity_IDs and http://wiki.vg/Entities#Mobs), better fix them before doing anything.

**Comment by rom1504** [comment]
* [ ] Enums of http://wiki.vg/Protocol in general (the non packet table), a bit like instruments.json (http://wiki.vg/Block_Actions)

**Comment by Gjum** [comment]
- [ ] http://wiki.vg/Protocol#Particle

**Comment by Gjum** [comment]
I came across another possible source of data http://technical-minecraft.wikia.com/

**Comment by Gjum** [comment]
[This page](http://minecraft.gamepedia.com/Block_entity) lists containers that store items. This can be presented in `windows.json` as a `persistent` flag to indicate if an item that is deposited in there will drop when the window is closed or if it will still be in there when the container is reopened.

Also some windows send a [Number Of Slots](http://wiki.vg/Protocol#Open_Window) = 0 when getting opened, which is probably related to this `persistent` flag.
